frame/session/src/historical/mod.rs

```diff
@@ -32,55 +32,71 @@ use sp_runtime::KeyTypeId;
 use sp_runtime::traits::{Convert, OpaqueKeys};
 use sp_session::{MembershipProof, ValidatorCount};
 use frame_support::{
-	decl_module, decl_storage, Parameter, print,
+	Parameter, print,
 	traits::{ValidatorSet, ValidatorSetWithIdentification},
 };
 use sp_trie::{MemoryDB, Trie, TrieMut, Recorder, EMPTY_PREFIX};
 use sp_trie::trie_types::{TrieDBMut, TrieDB};
-use super::{SessionIndex, Module as SessionModule};
+use super::{SessionIndex, Pallet as SessionModule, pallet as pallet_session};
+pub use pallet::*;
 
 mod shared;
 pub mod offchain;
 pub mod onchain;
 
-/// Config necessary for the historical module.
-pub trait Config: super::Config {
-	/// Full identification of the validator.
-	type FullIdentification: Parameter;
-
-	/// A conversion from validator ID to full identification.
-	///
-	/// This should contain any references to economic actors associated with the
-	/// validator, since they may be outdated by the time this is queried from a
-	/// historical trie.
-	///
-	/// It must return the identification for the current session index.
-	type FullIdentificationOf: Convert<Self::ValidatorId, Option<Self::FullIdentification>>;
-}
-
-decl_storage! {
-	trait Store for Module<T: Config> as Session {
```

Comment:
This is a breaking change. The historical pallet will no longer use the `Session` storage.

This needs a storage migration.